packages/dds/tree/docs/SharedTree Philosophy.md

```diff
@@ -0,0 +1,254 @@
+    This includes compatibility between different applications or versions of an application, scalability, availability, merge quality, etc.
+    More concretely, if faced with a new application collaboration requirement, it should be practical to either build it on top of SharedTree or extend SharedTree to support it.
+    Doing this should not require major reworking of either the application or SharedTree implementations, not require extending it in a way that the upstream SharedTree would not be willing to maintain.
+
+-   Generalize investments into improved user experiences across many different applications.
+
+    As much as possible, SharedTree should serve as a framework for sharing generalized collaborative logic, and a good default library of such logic.
+    For example merge resolution strategies, compatibility adapters, debug tools, common schema etc, can be generalized for easy use by all users of SharedTree.
+    Users of SharedTree can start by authoring their own specialized version of any of these (if needed), and then consider generalizing them as either independent libraries or upstream contributions.
+
+Another way to describe these goals is that SharedTree needs to be able to be **adopted with confidence**.
+Once adopted, it must **compatible with existing systems**, **easily learnable** and **productive** for developers new to collaborative application while being **flexible** in what it supports and produce **robust** and **high quality** collaborative experiences.
+
+In addition to these overall goals, SharedTree also needs to provide an MVP early in the development and internally be **maintainable** and **extensible**.
+This allows it to deliver value early, but also continue to be developed to reduce the development effort required of application authors to improve the experience it enables for end users.
+These internal requirements mostly follow from the above flexibility requirement, but are worth the extra emphasis.
+
+# Balancing Conflicting Goals
+
+These goals are values which the SharedTree tries to maximize.
+There are however cases where trade-offs must be made.
+For example, being compatible with existing libraries and conventions from non-collaborative applications can lead to APIs that lead to common collaboration pitfalls, like unclear merge behavior due to insufficient semantics captured through the editing APIs.
+Much of the design of SharedTree is to avoid having to make these tradeoffs and this is the preferred approach, however some can not be avoided, and they will be evaluated carefully.
+
+# Implications of these Goals
+
+Commonly desired collaboration features should be easy for applications to support, but when supporting them requires extra work, they can be explicitly opted out of or deferred to a future version.
+
+This include features like:
+
+-   cross version collaboration support to enable incremental rollout of updates.
```

Comment:
I think history (commonly asked for) belongs in this list

Reply:
I don't think supporting history needs to require extra work from the application (they don't need to provide any policy logic, extra services, etc), and its probably going to be opt in (if its a tree feature at all) not opt out, so I don't think it belongs in this list.